LocustScripts/commcarehq-referrals-search-for-beds.py

```diff
@@ -0,0 +1,325 @@
+                                         checkValue=self.FUNC_CREATE_PROFILE_AND_REFER_FORM_SUBMIT[
+                                             'submitResponseMessage'])
+
+    def _formplayer_post(self, command, extra_json=None, name=None, checkKey=None, checkValue=None, checkLen=None):
+        json = {
+            "app_id": self.build_id,
+            "domain": self.user.domain,
+            "locale": "en",
+            "restoreAs": self.user.login_as,
+            "username": self.user.username,
+        }
+        if extra_json:
+            json.update(extra_json)
+        name = name or command
+
+        if 'XSRF-TOKEN' not in self.client.cookies:
+            response = self.client.get(f"{self.parent.formplayer_host}/serverup")
+            response.raise_for_status()
+
+        xsrf_token = self.client.cookies['XSRF-TOKEN']
+        headers = {'X-XSRF-TOKEN': xsrf_token}
+        self.client.headers.update(headers)
+
+        with self.client.post(f"{self.user.formplayer_host}/{command}/", json=json, name=name,
+                              catch_response=True) as response:
+            data = response.json()
+            # logging.info("data-->" + str(data))
+            if "notification" in data and data["notification"]:
+                if data["notification"]["type"] == "error":
+                    logging.info("ERROR::-" + data["notification"]["message"] + ": With json" + str(json))
+                    response.failure("exception error--" + data["notification"]["message"])
+            if "exception" in data:
+                logging.info("ERROR::exception error--" + data['exception'])
+                logging.info("ERROR::user-info::" + self.user.username + "::" + self.user.login_as)
+                response.failure("exception error--" + data['exception'])
+            elif checkKey and checkKey not in data:
+                logging.info("error::" + checkKey + " not in data")
+                response.failure("ERROR::" + checkKey + " not in data")
+            elif checkKey and checkLen:
+                if len(data[checkKey]) != checkLen:
+                    logging.info("ERROR::len(data['" + checkKey + "']) != " + checkLen)
+                    response.failure("error::len(data['" + checkKey + "']) != " + checkLen)
+            elif checkKey and checkValue:
+                if data[checkKey] != checkValue:
+                    logging.info("ERROR::data['" + checkKey + "'] != " + checkValue)
+                    response.failure("error::data['" + checkKey + "'] != " + checkValue)
+        return response.json()
+
+
+class LoginCommCareHQWithUniqueUsers(HttpUser):
+    tasks = [WorkloadModelSteps]
+
+    formplayer_host = "/formplayer"
+    project = 'bha-referrals-perf'  # str(os.environ.get("project"))
+    domain_user_credential_force = str(os.environ.get("user_credential"))
+    app_config_force = str(os.environ.get("app_config"))
+    wait_time_force = "test"
```

Comment:
these could be added as custom command line args: https://docs.locust.io/en/stable/extending-locust.html#custom-arguments

Reply:
[cc98536](https://github.com/dimagi/dimagi-qa/pull/323/commits/cc98536d4f5652ee2d5c370a90c955f67f0d2b60)